### PR TITLE
Form.SerFieldValues method

### PR DESCRIPTION
**Pull request by YehudahA**
Add Form.SerFieldValues method, and change Form.Submit to use this method.
This method set input and select elements.

### CODE REVIEW

File: AngleSharp/Extensions/ApiExtensions.cs [hunk 1]
```diff
@@ -27,45 +28,35 @@ public static TElement CreateElement<TElement>(this IDocument document)
             where TElement : IElement
         {
             if (document == null)
-            {
```
Comment (FlorianRappl) [inline AngleSharp/Extensions/ApiExtensions.cs:30]: Do not delete these braces. Single statements won't be allowed in the whole codebase and every part that still uses single-statements is currently changed.

File: AngleSharp/Extensions/ApiExtensions.cs [hunk 2]
```diff
@@ -27,45 +28,35 @@ public static TElement CreateElement<TElement>(this IDocument document)
...
-                    {
-                        var element = (TElement)obj;
-                        var baseElement = element as Element;
+                for (var i = 0; i < parameters.Length; i++)
+                {
+                    var isDocument = parameters[i].ParameterType == typeof(Document);
+                    arguments[i] = isDocument ? document : parameters[i].DefaultValue;
+                }
 
-                        if (baseElement != null)
```
Comment (FlorianRappl) [inline AngleSharp/Extensions/ApiExtensions.cs:61]: From which version did you fork? Deleting these lines will result in failing tests. I recommend you to fork from the latest version and the prefer the changeset from the upstream when merging.

**Comment by FlorianRappl** [comment]
Couple of things that need to be fixed before this can be accepted:
- Use the latest version from the upstream. Your changes would remove a lot of necessary changes that should stay in the codebase.
- Please provide unit tests. These tests need to cover the case(s) that can't be handled with the current codebase, however, can be handled with your changes.

**Comment by YehudahA** [comment]
@FlorianRappl I'm sorry, you're absolutely right.
I re-commited my changes.

**Comment by FlorianRappl** [comment]
Looks much better - thanks. Could you also supply a few unit tests for that? Otherwise, the issue you mentioned won't be tested and there could be regression on this feature in the future.

Much appreciated! (sorry to be so picky :stuck_out_tongue_winking_eye:)

**Comment by YehudahA** [comment]
@FlorianRappl, I will do it.
But I don't find any unit test for the original form.submit(fields) method :(

**Comment by FlorianRappl** [comment]
One more reason to supply unit tests now :wink:.

**Comment by YehudahA** [comment]
- I changed the code to support radio inputs [set input.IsChecked instead of set Input.Value]
- I wrote some unit tests.

**Comment by FlorianRappl** [comment]
Very nice. Will be merged after v0.9.4 has been released.

**Pull request merged**